### DISCUSSION TITLE
How to automate an advance zone deployment

### DISCUSSION THREAD

**Discussion by KuasarCloud** [apache/cloudstack#9978] [Q&A] [answered]
Hi,
I created a all-in-one ACS deploy using Ansible and would like also to automate an advanced zone deployment and I was thinking in options
 - Using CloudMonkey but at some point I need to register a access/secret key which is manual, 
 - Another alternative I found on the code tools/marvin a way to automate it but not so sure if it going to work without any credentials
 - Deploy one manually and backup the database and restore the sql
 
 is there another option currently?
 
 BR

**Comment by weizhouapache** [comment, marked as answer]
> Using CloudMonkey but at some point I need to register a access/secret key which is manual,

if you use the default credential (admin/password), you can set it in .cmk/config file.
in this case, api key/secret key is not required.

> Another alternative I found on the code tools/marvin a way to automate it but not so sure if it going to work without any credentials

yes, it should work on the management server itself. 
it is used in the simulutor ci. you can refer to 

  https://github.com/apache/cloudstack/blob/8a2c0f3f732c39f222373f98e22f81d103f5b11f/.github/workflows/ci.yml#L285-L294


> Deploy one manually and backup the database and restore the sql

this is also ok for testing , especially if you want to test the cloudstack upgrade.

**Reply by KuasarCloud** [reply]
Thanks Wei, I will take a look!
